daemon/logger/adapter.go

```diff
@@ -0,0 +1,135 @@
+package logger
+
+import (
+	"encoding/binary"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/api/types/plugins/logdriver"
+	"github.com/docker/docker/pkg/plugingetter"
+	protoio "github.com/gogo/protobuf/io"
+	"github.com/pkg/errors"
+)
+
+// pluginAdapter takes a plugin and implements the Logger interface for logger
+// instances
+type pluginAdapter struct {
+	driverName   string
+	id           string
+	plugin       logPlugin
+	fifoPath     string
+	stream       io.WriteCloser
+	capabilities Capability
+	mu           sync.Mutex // synchronize access to the log stream
```

Comment:
Haha, the comment doesn't help here. I saw that it's like cool to place mutex on top of protected fields and separate that block by empty spaces.